ddtrace/contrib/internal/futures/threading.py

```diff
@@ -12,6 +14,7 @@ def _wrap_submit(func, args, kwargs):
     """
     # DEV: Be sure to propagate a Context and not a Span since we are crossing thread boundaries
     current_ctx: Optional[Context] = ddtrace.tracer.current_trace_context()
+    llmobs_ctx: Optional[Context] = core.dispatch_with_results("threading.submit", ()).llmobs_ctx.value
```

Comment:
a thought, and this may be out of scope for our team to address, but i wonder if we could decouple this more.

could we generically dispatch this event, and gather all the results into a list, and pass that list into the wrapped execution? then, in `_wrap_execution`, do `core.dispatch("threading.execution", ...)` over all the context objects passed in? 

this could be a moot point because maybe llmobs is the only other product that wants to insert itself here/use this signal, so it's not worth the effort to generalize. curious to know your thoughts though!

Reply:
I think if we were to have more use cases to inject into the threading module this would be a great idea. However as you mentioned currently it's just us trying to use this signal so I don't think it is optimal to decouple just for us, at least until another product finds a use we should look into decoupling then. Is that cool with you?

Reply:
yep that's cool w me, thanks!